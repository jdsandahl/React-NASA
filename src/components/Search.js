import React, { useState } from "react";
import PropTypes from 'prop-types';
import "../styles/Search.css";
import getImages from "../requests/getImages.js";

const Search = ({ setSearchResults }) => {
  const [searchValue, setSearchValue] = useState([]);
  const [loading, setLoading] = useState(false);

  const handleSubmit = async (event) => {
    setLoading(true);
    event.preventDefault();
    setSearchResults(await getImages(searchValue));
    setLoading(false);
  };

  return (
    <div className="Search" data-testid="Search">
      <form className="search-form" onSubmit={handleSubmit} data-testid="search-form">
        <input
          className="search-input"
          type="text"
          placeholder="Search"
          onChange={(e) => setSearchValue(e.target.value)}
          data-testid="search-input"
        />
        <button className="search-btn" type="submit" data-testid="search-btn">
          Go!
        </button>
      </form>
      {loading && (
        <div className="search-loader">
          <div className="loader" />
        </div>
      )}
    </div>
  );
};

Search.propTypes = {
    setSearchResults: PropTypes.func.isRequired,
}
export default Search;
