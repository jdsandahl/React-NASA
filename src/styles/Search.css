.Search {
  display: flex;
  flex-direction: column;
  align-content: center;
  justify-content: center;
}

.search-form {
    display: flex;
    flex-direction: row;
    justify-content: center;
}

.search-input {
  width: 30%;
  height: 50px;
  border-radius: 11px 0px 0px 11px;
  font-size: 20px;
  padding: 0 10px;
}

.search-btn {
    border: none;
    border-radius: 0px 11px 11px 0px;
    background-image: linear-gradient(to bottom right, #ec0037, #ad032b);
    color: #ffffff;
    font-size: 20px;
}

.search-btn:active {
    background-image: linear-gradient(to bottom right, #d3cfd0, #5f5e5e);   
}

.search-loader {
  display: flex;
  justify-content: center;
  padding: 10px;
}

.loader{
  border: 10px solid #f3f3f3; /* Light grey */
  border-top: 10px solid #ad032b; /* Blue */
  border-radius: 50%;
  width: 70px;
  height: 70px;
  animation: spin 2s linear infinite;
}

@keyframes spin {
  0% { transform: rotate(0deg); }
  100% { transform: rotate(360deg); }
}