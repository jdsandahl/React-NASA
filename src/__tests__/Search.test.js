import React from "react";
import { render } from "@testing-library/react";
import Search from "../components/Search.js";

describe("Search", () => {
  const { asFragment } = render(<Search />);

  test("renders correctly", () => {
    expect(asFragment()).toMatchSnapshot();
  });
});
