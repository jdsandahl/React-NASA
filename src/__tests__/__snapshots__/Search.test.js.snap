// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search renders correctly against snapshot 1`] = `
<DocumentFragment>
  <div
    class="Search"
    data-testid="Search"
  >
    <form
      class="search-form"
      data-testid="search-form"
    >
      <input
        class="search-input"
        data-testid="search-input"
        placeholder="Search"
        type="text"
      />
      <button
        class="search-btn"
        data-testid="search-btn"
        type="submit"
      >
        Go!
      </button>
    </form>
  </div>
</DocumentFragment>
`;
